rollup/core/classes/RollupLimits.cls

```diff
@@ -0,0 +1,57 @@
+public without sharing class RollupLimits {
+  @TestVisible
+  private static Integer stubbedQueryRows;
+
+  public class Tester {
+    private final transient RollupControl__mdt control;
+    public Tester(RollupControl__mdt control) {
+      this.control = control;
+    }
+
+    public Boolean hasExceededQueryNumberLimit {
+      get {
+        return this.control?.MaxNumberOfQueries__c < Limits.getQueries();
+      }
+    }
+    public Boolean hasExceededQueryRowLimit {
+      get {
+        return this.getRemainingQueryRows() < 0;
+      }
+    }
+    public Boolean hasExceededHeapSizeLimit {
+      get {
+        return (Limits.getLimitHeapSize() / 2) < Limits.getHeapSize();
+      }
+    }
+    public Boolean hasExceededDMLRowLimit {
+      get {
+        return this.control?.MaxParentRowsUpdatedAtOnce__c < Limits.getDmlRows();
+      }
+    }
+    public Boolean hasExceededCPUTimeLimit {
+      get {
+        Integer intervalTillTimeout = ((System.isBatch() || System.isQueueable()) && Test.isRunningTest() == false) ? 14000 : 3000;
```

Comment:
Where do the 14000 & 3000 interval values come from? Might be worth storing them as variables with descriptive names or adding a short comment to explain the rationale behind the values.

Reply:
They’re sentinel values for sync/async. I can make descriptive variables for them (but also may defer this change to next version, as these numbers have been there for a while)

Reply:
Yeah, for sure, I don't think this is a big deal & could wait until a future release

Reply:
I ended up having the time to fix this! Thanks for noticing, I should have converted to explanatory variables for this a while back